File: broadcastreceiverlibrary/src/main/java/com/osapps/broadcastreceiverlibrary/sms/exceptions/SmsException.java
package com.osapps.broadcastreceiverlibrary.sms.exceptions;

/**
 * Created by osapps on 09/05/2018.
 */
public abstract class SmsException extends RuntimeException {

    @Override
    public String getMessage() {
        return getErrorMsg();
    }

    protected abstract String getErrorMsg();

}

File: broadcastreceiverlibrary/src/main/java/com/osapps/broadcastreceiverlibrary/sms/exceptions/SmsException.kt
package com.osapps.broadcastreceiverlibrary.sms.exceptions

/**
 * Created by osapps on 09/05/2018.
 */
abstract class SmsException : RuntimeException() {

    protected abstract val errorMsg: String
    override val message: String?
        get() = errorMsg

}
